RecoEgamma/EgammaHFProducers/plugins/HFEMClusterProducer.cc

```diff
@@ -5,11 +5,30 @@
 // $Id: HFEMClusterProducer.cc,v 1.2 2007/09/19 Kevin Klapoetke
 //
 
-#include <iostream>
 #include "DataFormats/Common/interface/Handle.h"
-#include "FWCore/Framework/interface/ESHandle.h"
```

Comment:
Why are you removing this?

Reply:
If one includes `EventSetup.h`, one automatically gets the `ESHandle` because the `EventSetup` has functions that return `ESHandles`.

https://github.com/cms-sw/cmssw/blob/master/FWCore/Framework/interface/EventSetup.h

That means including `ESHandle.h` in plugin sources is generally unnecessary.

I had this idea to remove it after I saw some other ES migration PRs systematically remove includes of `ESHandle.h` let me see if I can find there PRs again to have a precedent to point to.

The same logic in principal also applies to `Handle.h` by the way.

Reply:
since this file uses ESHandle, it should be included directly and not rely on something else to include it indirectly.

Reply:
Okay, then I'll just change the file so it doesn't use the ESHandle anymore